### ISSUE TITLE
A potpourri of observations, bugs with and without fixes from me playing this

### ISSUE THREAD

**Issue by SomeTroglodyte** [MidnightPlugins/Lost-in-Midnight#39]
... excellent plugin.

IMHO what makes this excellent is that most of if just "fits" well into the ES universe - or is separate enough to ignore. :smile_cat: :rocket: 

That said, here's a very unordered list of notes from me comparing a 2024-02-24 release download that I had patched to my pleasure with today's main branch:
- Still too many dependencies on Midnight Scrapyard - those the other way round are much less and easily fixable - you probably already know.
- Do read errors.txt please - same as console output when run from a shell. I got the errors down to very few - those I can't fix mentioned here.
- Can't fix for example: `(): Defaulting missing "drag" attribute to 100.0, has outfits:` - no outfits listed. Can't find which ship definition if any trips this - no name, no outfits, no nothing... But the message is there with saves folder empty and only your plugin active, and not after disabling your plugin. I even patched ES to log missing baseAttributes.drag in Ship::Load and none was relevant... I believe there's a ES bug that it somehow loads a ship template where there's no ship keyword, but that's just a hunch. May be an empty-looking but actually containing tabs line somewhere...
- Some missions use assignment instead of comparison on their conditions - it should be `"year" == 3014` not `"year" = 3014` - five instances. See patch below.
- The "lone pixel" error message refers to the Pug paragon maboro ship sprite (I needed a patch in ES's Mask.cpp to even find this).  See zip below for a replacement that doesn't trip the error message.
- Geddonian World Crusher or crystal sentries give no combat rating - See patch below.
- Having frames 2 and 4 of the Geddonian World Crusher empty - ES doesn't like. Giving those a copy of frame 0 with alpha = 25% looks fine. ("Unable to create mask for 230x220 px image: all pixels were transparent") -  See zip below.
- Missing sounds - photonic and meteor-ver8 still missing, new: plasma-ver8 missing too.
- Missing thumbnails - most seem fixed, though I would have gone the other way: ES vanilla normally has identical names under ships and under thumbnails. Most is done just fine - so keep your naming conventions :+1:
- But: Missing image "thumbnail/modified flivver-thumb".
- Invalid Trailblazer loadout - already fixed :+1: 
- "Quantum Turle Blade" - not yet fixed - See patch below.
- mission "Thraal: Moreau (Failed)" - I had better experience uncommenting the `not "Thraal: Moreau (Cleanup): declined"` line
- invalid npcs in mission "Caspar Delivery" - already fixed :+1: - that one took me a few minutes to see the problem, but just updating earlier would have saved me the effort.
- Korbschläger - the umlaut does not display anywhere on Linux - just FYI, no idea what to do about that.
- Double deadline on mission "Midnight Earth Retirement Inheritance" makes it unwinnable - or maybe the `not planet Earth` you added fixes what I experienced? Maybe related to #21? I played it with `deadline 1` removed - just fine.
- Map Paeli Patch : "planet/browndwarf-t-rouge" was an upstream mistake you copied - it needs to be "planet/browndwarf-t-rogue" - See patch below.
- mission "Sheragi Shenanigans: Counterfeits" - I didn't get it though I believed it was intended to - I needed a `has "Sheragi Shenanigans: Sapphire Shield 4: declined"` on the prerequisites to play it.
- Image ship/cloaked is complained about by ES, tried giving it a 55% alpha black one, didn't help - so - maybe ES will require a completely black tiny square to stop complaining? Or leave the error...
- Lone image "pirate scrounger@sw.png" in the thumbnails folder - that belongs in ship, is there, so delete it - See patch below.
- Outfit "Test Drive" on the "Starcadia (Prototype)" does not exist - no idea what you planned.
- Missing image "outfit/spirit board" - I've gimped one from some duckduck image search... - See zip below.
- Missing image "ship/station5ver8" -> use "planet/station5ver8" instead (ship "Titania Station")? - See patch below.

The images I mentioned are in this zip:
[Lost in Midnight - images patch.zip](https://github.com/MidnightPlugins/Lost-in-Midnight/files/15287176/Lost.in.Midnight.-.images.patch.zip)
You're free to use as you see fit w/o attribution - but sorry I haven't kept any upstream info where I took that spirit board from, so that probably won't do as-is.

The textual changes I mentioned as git patch - I can do a PR if you want, but the scope is small, so just copying or ignoring as you see fit is OK too. Some are just a suggestion like giving cost to the worldcrusher so trouncing it gives XP... And it's better you read them individually anyway.

<details>

```patch
Index: data/drak midnight/Geddonian Worldcrusher.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/drak midnight/Geddonian Worldcrusher.txt b/data/drak midnight/Geddonian Worldcrusher.txt
--- a/data/drak midnight/Geddonian Worldcrusher.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/drak midnight/Geddonian Worldcrusher.txt	(date 1715531237052)
@@ -82,6 +82,7 @@
 outfit "Graviton Channeler"
 	category "Turrets"
 	thumbnail outfit/unknown
+	"cost" 1e8
 	"mass" 54
 	"outfit space" -54
 	"weapon capacity" -54
@@ -112,6 +113,7 @@
 outfit "Quantum Paradox Diffusion"
 	category "Turrets"
 	thumbnail outfit/unknown
+	"cost" 1e8
 	"mass" 500
 	"outfit space" -500
 	"weapon capacity" -500
@@ -135,6 +137,7 @@
 outfit "Quantum Paradox Inversion"
 	category "System"
 	thumbnail outfit/unknown
+	"cost" 1e8
 	"mass" 180
 	"outfit space" -180
 	"hull repair rate" 18
@@ -147,6 +150,7 @@
 outfit "Quantum Paradox Propulsion"
 	category "Engines"
 	thumbnail outfit/unknown
+	"cost" 1e8
 	"mass" 270
 	"outfit space" -270
 	"engine capacity" -270
@@ -179,6 +183,7 @@
 outfit "Imaginary Number Fusion Matrix"
 	category "Power"
 	thumbnail outfit/unknown
+	"cost" 1e8
 	"mass" 729
 	"outfit space" -729
 	"energy generation" 180
@@ -194,6 +199,7 @@
 outfit "Omni-Directional Photonic Emitter"
 	category "Turrets"
 	thumbnail outfit/unknown
+	"cost" 1e7
 	"mass" 54
 	"outfit space" -54
 	"weapon capacity" -54
@@ -222,6 +228,7 @@
 outfit "Space Differential Stabilizer"
 	category "Turrets"
 	thumbnail outfit/unknown
+	"cost" 1e7
 	"mass" 45
 	"outfit space" -45
 	"weapon capacity" -45
@@ -245,6 +252,7 @@
 outfit "Zero Division Core"
 	category "Power"
 	thumbnail outfit/unknown
+	"cost" 1e7
 	"mass" 73
 	"outfit space" -73
 	"energy generation" 90
@@ -268,9 +276,11 @@
 		category "Heavy Warship"
 		licenses
 			Author
+		"cost" 1e10
 		"shields" 999999
 		"hull" 999999
 		"automaton" 1
+		"crew equivalent" 16000
 		"mass" 9999
 		"drag" 45.0
 		"heat dissipation" 1.8
@@ -374,9 +384,11 @@
 		category "Drone"
 		licenses
 			Author
+		"cost" 4e8
 		"shields" 99999
 		"hull" 99999
 		"automaton" 1
+		"crew equivalent" 800
 		"mass" 999
 		"drag" 9.0
 		"heat dissipation" 1.8
@@ -496,4 +508,4 @@
 		on kill
 			set "Geddonian Worldcrusher: done"
 			clear "Geddonia Worldcrusher: offered"
-			fail
\ No newline at end of file
+			fail
Index: data/human midnight/Missions Sunracer.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/human midnight/Missions Sunracer.txt b/data/human midnight/Missions Sunracer.txt
--- a/data/human midnight/Missions Sunracer.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/human midnight/Missions Sunracer.txt	(date 1715532225990)
@@ -1996,7 +1996,7 @@
 							has "outfit (flagship installed): Quantum Phoenix Blade"
 							has "outfit (flagship installed): Quantum Snake Blade"
 							has "outfit (flagship installed): Quantum Tiger Blade"
-							has "outfit (flagship installed): Quantum Turle Blade"
+							has "outfit (flagship installed): Quantum Turtle Blade"
 					goto keyblade
 				`	(Run.)`
 					to display
@@ -2239,4 +2239,4 @@
 		"racing position tertio" >?= 6
 		conversation
 			scene "thumbnail/modified flivver-thumb"
-			`When you leave the <ship> there is a camera crew waiting for you. A Lionheart representative arrives with the Accelerator in tow. They make a big show of publicly registering it to you. The photo op ends with the representative wishing you good luck on your next race.`
\ No newline at end of file
+			`When you leave the <ship> there is a camera crew waiting for you. A Lionheart representative arrives with the Accelerator in tow. They make a big show of publicly registering it to you. The photo op ends with the representative wishing you good luck on your next race.`
Index: data/remnant midnight/Missions Hunting Beelzebub.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/remnant midnight/Missions Hunting Beelzebub.txt b/data/remnant midnight/Missions Hunting Beelzebub.txt
--- a/data/remnant midnight/Missions Hunting Beelzebub.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/remnant midnight/Missions Hunting Beelzebub.txt	(date 1715532863508)
@@ -40,10 +40,7 @@
 			has "Remnant: Deep Surveillance: failed"
 			has "Remnant: Scanning Tolerances: done"
 		or
-			has "event: kestrel available: more bays"
-			has "event: kestrel available: more engines"
-			has "event: kestrel available: more shields"
-			has "event: kestrel available: more weapons"
+			has "kestrel available"
 		or
 			and
 				not "ignored paris"
Index: data/sheragi midnight/Missions Sheragi Shenanigans.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/sheragi midnight/Missions Sheragi Shenanigans.txt b/data/sheragi midnight/Missions Sheragi Shenanigans.txt
--- a/data/sheragi midnight/Missions Sheragi Shenanigans.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/sheragi midnight/Missions Sheragi Shenanigans.txt	(date 1715532475679)
@@ -1085,7 +1085,7 @@
 					has "outfit (flagship installed): Quantum Phoenix Blade"
 					has "outfit (flagship installed): Quantum Snake Blade"
 					has "outfit (flagship installed): Quantum Tiger Blade"
-					has "outfit (flagship installed): Quantum Turle Blade"
+					has "outfit (flagship installed): Quantum Turtle Blade"
 			label padlock
 			``
 			`There's a chain with a combination padlock. Very low tech.`
Index: data/human midnight/Space Stations Human.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/human midnight/Space Stations Human.txt b/data/human midnight/Space Stations Human.txt
--- a/data/human midnight/Space Stations Human.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/human midnight/Space Stations Human.txt	(date 1715534918011)
@@ -126,7 +126,7 @@
 
 
 ship "Titania Station"
-	sprite "ship/station5ver8"
+	sprite "planet/station5ver8"
 		"scale" 4
 	"never disabled"
 	swizzle 0
Index: data/quarg midnight/Missions Midnight Lost Orphan.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/quarg midnight/Missions Midnight Lost Orphan.txt b/data/quarg midnight/Missions Midnight Lost Orphan.txt
--- a/data/quarg midnight/Missions Midnight Lost Orphan.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/quarg midnight/Missions Midnight Lost Orphan.txt	(date 1715532475623)
@@ -1570,7 +1570,7 @@
 						has "outfit (flagship installed): Quantum Phoenix Blade"
 						has "outfit (flagship installed): Quantum Snake Blade"
 						has "outfit (flagship installed): Quantum Tiger Blade"
-						has "outfit (flagship installed): Quantum Turle Blade"
+						has "outfit (flagship installed): Quantum Turtle Blade"
 
 			label alone
 			`You are suddenly lifted off the ground by a supernatural force. "I knew you'd come back, Captain <first> <last>."`
@@ -1587,7 +1587,7 @@
 					has "outfit (flagship installed): Quantum Phoenix Blade"
 					has "outfit (flagship installed): Quantum Snake Blade"
 					has "outfit (flagship installed): Quantum Tiger Blade"
-					has "outfit (flagship installed): Quantum Turle Blade"
+					has "outfit (flagship installed): Quantum Turtle Blade"
 
 			`You've sent T.H.O.R. to scout ahead. After your previous experience here, you aren't taking any chances.`
 			choice
Index: data/hai midnight/Missions Quantum Key Blade.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/hai midnight/Missions Quantum Key Blade.txt b/data/hai midnight/Missions Quantum Key Blade.txt
--- a/data/hai midnight/Missions Quantum Key Blade.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/hai midnight/Missions Quantum Key Blade.txt	(date 1715531704952)
@@ -627,7 +627,7 @@
 			has "outfit (flagship installed): Quantum Phoenix Blade"
 			has "outfit (flagship installed): Quantum Snake Blade"
 			has "outfit (flagship installed): Quantum Tiger Blade"
-			has "outfit (flagship installed): Quantum Turle Blade"
+			has "outfit (flagship installed): Quantum Turtle Blade"
 		or
 			and
 				has "flagship planet attribute: unfettered"
@@ -693,7 +693,7 @@
 			has "outfit (flagship installed): Quantum Phoenix Blade"
 			has "outfit (flagship installed): Quantum Snake Blade"
 			has "outfit (flagship installed): Quantum Tiger Blade"
-			has "outfit (flagship installed): Quantum Turle Blade"
+			has "outfit (flagship installed): Quantum Turtle Blade"
 		or
 			and
 				has "flagship planet attribute: unfettered"
@@ -759,7 +759,7 @@
 			has "outfit (flagship installed): Quantum Phoenix Blade"
 			has "outfit (flagship installed): Quantum Snake Blade"
 			has "outfit (flagship installed): Quantum Tiger Blade"
-			has "outfit (flagship installed): Quantum Turle Blade"
+			has "outfit (flagship installed): Quantum Turtle Blade"
 		or
 			and
 				has "flagship planet attribute: unfettered"
Index: data/human midnight/Missions North Wind Hai Power.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/human midnight/Missions North Wind Hai Power.txt b/data/human midnight/Missions North Wind Hai Power.txt
--- a/data/human midnight/Missions North Wind Hai Power.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/human midnight/Missions North Wind Hai Power.txt	(date 1715532475647)
@@ -838,7 +838,7 @@
 							has "outfit (flagship installed): Quantum Phoenix Blade"
 							has "outfit (flagship installed): Quantum Snake Blade"
 							has "outfit (flagship installed): Quantum Tiger Blade"
-							has "outfit (flagship installed): Quantum Turle Blade"
+							has "outfit (flagship installed): Quantum Turtle Blade"
 					goto keyblade
 				`	(Use T.H.O.R.)`
 					to display
Index: data/human midnight/Jobs Misc.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/human midnight/Jobs Misc.txt b/data/human midnight/Jobs Misc.txt
--- a/data/human midnight/Jobs Misc.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/human midnight/Jobs Misc.txt	(date 1715534126936)
@@ -2702,7 +2702,7 @@
 		or
 			"year" < 3014
 			and
-				"year" = 3014
+				"year" == 3014
 				"month" < 6
 		random < 15
 	source
@@ -2729,7 +2729,7 @@
 		or
 			"year" < 3014
 			and
-				"year" = 3014
+				"year" == 3014
 				"month" < 6
 		random < 15
 	source
@@ -2913,7 +2913,7 @@
 		or
 			"year" < 3014
 			and
-				"year" = 3014
+				"year" == 3014
 				"month" < 5
 	on complete
 		"mining jobs done" ++
@@ -2935,7 +2935,7 @@
 		or
 			"year" < 3014
 			and
-				"year" = 3014
+				"year" == 3014
 				"month" < 5
 		random < 2
 	on complete
@@ -2965,7 +2965,7 @@
 		or
 			"year" < 3014
 			and
-				"year" = 3014
+				"year" == 3014
 				"month" < 5
 	on complete
 		"mining jobs done" ++
Index: data/Map Paeli Patch.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/data/Map Paeli Patch.txt b/data/Map Paeli Patch.txt
--- a/data/Map Paeli Patch.txt	(revision f1c2bc083b59bd79463c840a27913a262a7fe9df)
+++ b/data/Map Paeli Patch.txt	(date 1715532574195)
@@ -37,7 +37,7 @@
 	hazard "Ember Waste Base Heat" 100
 	hazard "Ember Waste Base Storm" 9000
 	object
-		sprite planet/browndwarf-t-rouge
+		sprite planet/browndwarf-t-rogue
 		period 10
 	object
 		sprite planet/desert4-b
diff --git a/images/thumbnail/pirate scrounger@sw.png b/images/thumbnail/pirate scrounger@sw.png
deleted file mode 100644
index d3e61a145c2671719b76ab10e1bbca121fde8829..0000000000000000000000000000000000000000
GIT binary patch
literal 0
Hc$@<O00001

```
</details>

Lastly: You seem to manage this entirely through the github web functions - in my experience such a repo can be maintained in the free android studio variant of jetbrains idea, just using its text editor, diff engine and git frontend - makes for much more readable commit histories... I can explain if you want. Or look under my repos for the scratchpad where I kept (sometimes outdated) notes how I use studio, maybe that helps.

**Comment by MidnightPlugins** [comment]
Thanks! A lot of issues is me putting missing files in the release but forgetting to add it to the repo.

Still too many dependencies on Midnight Scrapyard - There are some outfits where half the definition is in Scrapyard, but they should combine when running both plugins. Otherwise, I'm not sure what you mean.

Geddonian World Crusher or crystal sentries give no combat rating - I'm not sure what you mean by "give no combat rating"

Having frames 2 and 4 of the Geddonian World Crusher empty - Intentional; it was designed to take advantage of an error that made it invincible during frames where a ship had non-connected parts. That error has since been removed from vanilla. The blank frames are there to restore some of that invincibility.

mission "Thraal: Moreau (Failed)" - I'll update after testing to see if the error that failed to remove clearance from destination planet was fixed in the recent vanilla update

Double deadline on mission "Midnight Earth Retirement Inheritance" - Deadline by itself gives a deadline equal to the distance to the destination system. The extra deadline 1 should extend the deadline by one day in case the mission offered on Luna or Mars. Will test again to confirm this hasn't changed.

Map Paeli Patch - Intended to fix an error that no longer exists in vanilla; deleted the file

mission "Sheragi Shenanigans: Counterfeits" - Should offer after offered epilogue A; After some consideration I decided that there's no reason to make the player wait so added 4: declined as a possible pre req.

Image ship/cloaked - leaving error as is

Lone image "pirate scrounger@sw.png" in the thumbnails folder - Intentional for the shipyard thumbnail

Outfit "Test Drive" on the "Starcadia (Prototype)" - developer outfit; removed

Missing image "ship/station5ver8" - I put station images for all stations that spawn as ships in the ship folder because the game had problems with hit detection for some stations using the vanilla planet images; this is an example of an image placed in the release but overlooked in the repo
